vita3k/app/src/app.cpp

```diff
@@ -34,31 +34,44 @@ void error_dialog(const std::string &message, SDL_Window *window) {
     }
 }
 
-static const uint32_t frames_size = 20;
 void calculate_fps(EmuEnvState &emuenv) {
+    static const uint32_t frames_size = sizeof(emuenv.fps_values) / sizeof(emuenv.fps_values[0]);
+    static uint16_t deltaTimeAcc = 0; // delta time accumulator for the window title
     const uint32_t sdl_ticks_now = SDL_GetTicks();
-    const uint32_t ms = sdl_ticks_now - emuenv.sdl_ticks;
+    const uint32_t deltaTime = sdl_ticks_now - emuenv.sdl_ticks; // Delta time since last frame
+    deltaTimeAcc += deltaTime;
 
-    if (ms >= 1000 && emuenv.frame_count > 0) {
-        // Set Current FPS
-        // round division to nearest integer instead of truncating
-        const uint32_t frame_count = static_cast<std::uint32_t>(emuenv.frame_count);
-        emuenv.fps = (frame_count * 1000 + ms / 2) / ms;
-        emuenv.ms_per_frame = (ms + frame_count / 2) / frame_count;
-        emuenv.sdl_ticks = sdl_ticks_now;
-        emuenv.frame_count = 0;
-        set_window_title(emuenv);
+    if (emuenv.frame_count == 0)
+        return;
+
+    // Only process frames that last AT LEAST 1ms, else math breaks
+    // Becuase of this the FPS calculation is maxed out to 1000 FPS (not like any game is gonna hit that)
+    // This only happens on some rare cases when the command list is empty and the gpu has to do nothing
+    if (deltaTime >= 1) {
+        // Set FPS and MS
+        emuenv.ms_per_frame = deltaTime;
+        emuenv.fps = 1000 / deltaTime;
```

Comment:
why change calcule for get fps ? current code is full fine

Reply:
fps is how many frametimes can fit in 1000 milliseconds, so by doing basic math you can figure out the fps based on the delta time since the last frame, before it was something else